test/node-rpc-test.js

```diff
@@ -245,6 +262,41 @@ describe('RPC', function() {
       assert.equal(node.chain.tip.version, info.version);
     });
 
+    it('should rpc getblockbyheight', async () => {
+      // Getting block height of chain tip
+      const blockheight = node.chain.tip.height;
+      const blockHash = node.chain.tip.hash;
+
+      // verbose=0 details=0
+      let blockInfo;
+      blockInfo = await nclient.execute('getblockbyheight', [blockheight, 0, 0]);
+      const block = Block.fromRaw(Buffer.from(blockInfo, 'hex'));
+      assert.bufferEqual(blockHash, block.hash());
+
+      // verbose=1 details=0
+      blockInfo = await nclient.execute('getblockbyheight', [blockheight, 1, 0]);
+      const properties = [
+        'hash', 'confirmations', 'strippedsize',
+        'size', 'weight', 'height', 'version',
+        'versionHex', 'merkleroot', 'coinbase',
+        'tx', 'time', 'mediantime', 'nonce',
+        'bits', 'difficulty', 'chainwork',
+        'nTx', 'previousblockhash', 'nextblockhash'
+      ];
+      for (const property of properties)
+        assert(property in blockInfo);
+
+      assert.bufferEqual(blockHash, block.hash());
```

Comment:
I dont think the value of `block` has changed since line 274 where you already checked it?! Maybe this needs to be `blockInfo.hash` ? This goes for the next check on line 297 as well.